pkg/volume/flexvolume/flexvolume_util.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/util/exec"
+)
+
+const (
+	initCmd     = "init"
+	setUpCmd    = "setup"
+	tearDownCmd = "teardown"
+)
+
+const (
+	// StatusSuccess represents the successful completion of operation in
+	// the backend.
+	StatusSuccess = "Success"
+	// StatusFailed represents that the operation is failed in the backend.
+	StatusFailure = "Failed"
+)
+
+// FlexVolumeDriverStatus represents the return value of the driver callout.
+type FlexVolumeDriverStatus struct {
+	// Status of the callout. One of "Success" or "Failure".
+	Status string
+	// Message is the reason for failure.
+	Message string
+	/* Device assigned by the driver. */
+	Device string `json:"device"`
```

Comment:
What purpose does this `Device string` serve? Is it different than the `driverName` that is in the flexVolumeBuilder? If so, does it make sense to require all plugins to specify it in output?